openhands/controller/agent_controller.py

```diff
@@ -199,32 +196,42 @@ async def _react_to_exception(
                 err_id = 'STATUS$ERROR_LLM_AUTHENTICATION'
             self.status_callback('error', err_id, type(e).__name__ + ': ' + str(e))
 
-    async def start_step_loop(self):
-        """The main loop for the agent's step-by-step execution."""
-        self.log('info', 'Starting step loop...')
-        while True:
-            if not self._is_awaiting_observation() and not should_continue():
-                break
-            if self._closed:
-                break
-            try:
-                await self._step()
-            except asyncio.CancelledError:
-                self.log('debug', 'AgentController task was cancelled')
-                break
-            except Exception as e:
-                traceback.print_exc()
-                self.log('error', f'Error while running the agent: {e}')
-                await self._react_to_exception(e)
+    def step(self):
+        asyncio.create_task(self._step_with_exception_handling())
 
-            await asyncio.sleep(0.1)
+    async def _step_with_exception_handling(self):
+        try:
+            await self._step()
+        except Exception as e:
+            traceback.print_exc()
+            self.log('error', f'Error while running the agent: {e}')
+            reported = RuntimeError(
+                'There was an unexpected error while running the agent.'
+            )
```

Comment:
This was intended, the exception name will _not_ be displayed in the UI chat box, again? 😅

FWIW I think it's somewhat useful for debugging errors seen on the hosted version. It's just a clue, but it's a decent clue.

Reply:
Also, losing the underlying exception info implies that on [litellm.AuthenticationError](https://github.com/All-Hands-AI/OpenHands/pull/5868/files#diff-1e082f415afb40de1d5c2a3fa327326225e1da28a142ad89c9524109f35fb36eR195), it will not display there the user-friendly text we currently have.

Reply:
Yeah it's generally considered unsafe to propagate arbitrary errors down to the client, because they can leak sensitive information like passwords. I did add litellm.AuthenticationError though!